crmsh/ui_resource.py

```diff
@@ -497,11 +498,54 @@ def do_constraints(self, context, resource):
 
     @command.wait
     @command.completers(compl.resources, _attrcmds, compl.nodes)
-    def do_failcount(self, context, rsc, cmd, node, value=None):
+    def do_failcount(self, context, rsc, cmd, node, value=None, operation=None, interval=None):
         """usage:
-        failcount <rsc> set <node> <value>
+        failcount <rsc> set <node> <value> [operation] [interval]
         failcount <rsc> delete <node>
         failcount <rsc> show <node>"""
+        if cmd == "set":
+            nodeid = utils.get_nodeid_from_name(node)
+            if nodeid is None:
+                context.fatal_error("Node {} not in this cluster".format(node))
+
+            # query the current failcount status
+            query_cmd = "cibadmin -Ql --xpath '/cib/status/node_state[@id='{}']'".format(nodeid)
+            rc, out, err = utils.get_stdout_stderr(query_cmd)
+            if rc != 0:
+                context.fatal_error(err)
+
+            # try to get failcount dict {operation:interval}
+            import re
+            failcount_res = re.findall(r'fail-count-{}#(.*)_([0-9]+)'.format(rsc), out)
+            if not failcount_res:
+                context.fatal_error("No failcount on node {} for resource {}".format(node, rsc))
+            failcount_dict = dict(failcount_res)
+
+            # validate for operation and interval
+            if operation and operation not in failcount_dict.keys():
+                context.fatal_error("Should specify operation between \"{}\"".format(' '.join(failcount_dict.keys())))
+            if (operation and interval) and (operation, interval) not in failcount_res:
+                context.fatal_error("Should specify (operation, interval) between {}".format(failcount_res))
+
+            # just one failcount entry
+            if len(failcount_res) == 1:
+                operation = failcount_res[0][0]
+                interval = failcount_dict[operation]
+                rsc = '{}#{}_{}'.format(rsc, operation, interval)
+
+            # multiple failcount entries for this resource and node
+            if len(failcount_res) > 1:
+                if operation and interval:
+                    rsc = '{}#{}_{}'.format(rsc, operation, interval)
+                else:
+                    # using '-P' option of 'crm_attribute' command
+                    cmd = "set_p"
+                    if operation:
+                        op_interval_str = '|'.join(['{}_{}'.format(operation, inter) for op, inter in failcount_res if op==operation])
+                    else:
+                        op_interval_str = '|'.join(['{}_{}'.format(op, inter) for op, inter in failcount_res])
```

Comment:
Hrm, it works correctly only if  value 0 is specified, doesn't it? For example with a specified value 2, it'll set every matched fail-count entries to 2, which will eventually contribute `2 * n` to the resulting fail-count (sum), won't it? 

I think we still need to differentiate cases for 0 and non-zero values. And for non-zero cases, check how many fail-count entries are really matched.  Set it if only one is matched, otherwise drop an error asking user to be specific.

Reply:
@gao-yan 
1) I've added the transition between ms to sec and sec to ms;
2) For non-zero set to multiple entries, I've changed, drop an error for input specific (operation, interval)
Above both are mentioned in doc.
Thanks for suggestions!